collector/lib/CollectorConfig.cpp

```diff
@@ -400,6 +406,48 @@ void CollectorConfig::HandleSinspEnvVars() {
   }
 }
 
+void CollectorConfig::HandleConfigMapString(const std::string& jsonString) {
+  sensor::CollectorConfig config;
+  auto status = google::protobuf::util::JsonStringToMessage(jsonString, &config);
+
+  if (!status.ok()) {
+    CLOG(ERROR) << "Failed to parse runtime config. Error: " << status.message();
+    CLOG(ERROR) << "ConfigMap file contents: " << jsonString;
+  } else {
+    SetRuntimeConfig(config);
+    CLOG(INFO) << "Set the runtime config using a configmap";
+    CLOG(INFO) << config.DebugString();
+  }
+}
+
+std::string readJsonFileToString(const std::filesystem::path& filePath) {
+  std::ifstream fileStream(filePath);
+  if (!fileStream.is_open()) {
+    CLOG(WARNING) << "Unable to open ConfigMap file: " << filePath
+                  << ". File may not exist or permissions may be incorrect. "
+                  << "If a ConfigMap was not created intentionally, which is the default, "
+                  << "no action is needed.";
```

Comment:
This is way too verbose, I propose this function to be re-written to something like this:
```cpp
std::string readJsonFileToString(const std::filesystem::path& filePath) {
  if (!std::filesystem::exists(filePath) {
    CLOG(DEBUG) << "No configuration file found.";
    return "";
  }

  std::ifstream fileStream(filePath);
  if (!fileStream.is_open()) {
    CLOG(WARNING) << "Unable to open configuration file: " << filePath;
    return "";
  }

  std::string content((std::istreambuf_iterator<char>(fileStream)),
                      std::istreambuf_iterator<char>());

  if (content.empty()) {
    CLOG(WARNING) << "Configuration file is empty: " << filePath;
  }
  return content;
}
```

Also, as you may have noticed from the changes, don't refer to the configuration file as a `ConfigMap`, we are configuring it as a configmap in kubernetes, but we are reading a configuration file, not the configmap object.

Reply:
Alternatively, yaml-cpp can directly read files if we move to use that.

Reply:
I am now using yaml instead of json.